### PR TITLE
Write helm kubernetes yml files

### PR DESCRIPTION
**Pull request by lightvector**
Attempt to write yml files for kubernetes. Wooo!

@tychota - When you get the chance, please let me know what you think. I have a few questions about various things - the expected workflow for using these files, versioning and tagging and upload of images, how secrets work, how internal services communication and hostname works for redis and celery, but we can address these in an audio chat when you have time.

Redis is missing from this push, but I think mostly everything else is here aside from secrets?

### CODE REVIEW

**Comment by lightvector** [comment]
Also I'm not sure that this is the right place to put these - maybe they belong in another repo. But given that "docker compose" configuration lives in this same repo, and kubernetes yml is supposed to be a replacement for "docker compose", maybe it makes sense to put it here too? So that if we need to update it (for example django needs a new environment variable or changes in its start command) we can update both docker-compose and kubernetes in a synchronized way?

Not sure. This is a part of the workflow I am fuzzy on, let me know. :)

**Comment by tychota** [comment]
> Also I'm not sure that this is the right place to put these - maybe they belong in another repo. But given that "docker compose" configuration lives in this same repo, and kubernetes yml is supposed to be a replacement for "docker compose", maybe it makes sense to put it here too? So that if we need to update it (for example django needs a new environment variable or changes in its start command) we can update both docker-compose and kubernetes in a synchronized way?

I would say that they are more decoupled that docker-compose thought.

Eg sometimes docker-compose say `build: ./relative/path/to/dockerfile` but here we separate this into two opérations. Building image is not running them.

File: kubernetes/celery-beat/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: celery-beat-deployment
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:4]: [Importance 2/10]

Maybe we should create a namespace `namespace: website` and have the namespace created in another file.

Having some namespace makes it easy to say for instance thatpod inside  `website` should have a service mesh attached on.

File: kubernetes/celery-beat/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  labels:
+    app: celery-beat
+spec:
+  replicas: {{.Values.celeryBeat.replicaCount}}
+  strategy:
+    type: Recreate
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:10]: [Importance 5/10]

Recreate (in opposition to RollingUpdate) destroy all then restart all.
This way you effectivly cut the service for a few seconds. Is it expected ?

File: kubernetes/celery-beat/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      app: celery-beat
+  template:
+    metadata:
+      labels:
+        app: celery-beat
+    spec:
+      containers:
+      - name: celery-beat
+        image: {{.Values.django.image.repo}}:{{.Values.django.image.tag}}
+        command: [ "/entrypoint", "/start-celerybeat"]
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:22]: [Importance 2/10]

Entrypoint doesn't work in kubernetes (as you noticed).

Maybe we can rework images to have plain script instead of two scripts
Eg just

/start-celerybeat

File: kubernetes/celery-beat/deployment.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        envFrom:
+        - secretRef:
+            name: django-env-secrets
+        - secretRef:
+            # POSTGRES_USER and POSTGRES_PASSWORD
+            name: postgres-env-secrets
+        env:
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:29]: You can also use env from but from configmap, this time

https://kubernetes.io/docs/concepts/configuration/configmap/

File: kubernetes/celery-beat/deployment.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  template:
+    metadata:
+      labels:
+        app: celery-beat
+    spec:
+      containers:
+      - name: celery-beat
+        image: {{.Values.django.image.repo}}:{{.Values.django.image.tag}}
+        command: [ "/entrypoint", "/start-celerybeat"]
+        envFrom:
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:23]: Nice, that is exactly what i had in mind

File: kubernetes/celery-beat/deployment.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          value: {{.Values.database.dbPort}}
+        volumeMounts:
+        - name: django-fileserver-storage
+          mountPath: /data
+      - name: cloud-sql-proxy
+        image: {{.Values.database.cloudsqlimage}}
+        command:
+        - "/cloud_sql_proxy"
+        - "-ip_address_types=PRIVATE"
+        - "--instances={{.Values.database.instanceConnectionName}}=tcp:{{.Values.database.dbPort}}"
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:44]: Well done, you found this and this is an headache !

Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:44]: [Importance 2/10]

Idealy we should use configmap + env variable

File: kubernetes/celery-beat/deployment.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        securityContext:
+          runAsNonRoot: true
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:46]: [Importance 3/10]

Idealy we should go further on security, for instance using `readOnlyRootFilesystem: true`.

I can compare later what i do at work and give more precise recomendation

File: kubernetes/celery-beat/deployment.yml (hunk 4 above)
Comment (tychota) [reply kubernetes/celery-beat/deployment.yml:29]: In the mean time, some docs https://sysdig.com/wp-content/uploads/2019/01/kubernetes-security-guide.pdf

(chapter 4 and 5)

File: kubernetes/celery-beat/deployment.yml (hunk 7 above)
Comment (tychota) [reply kubernetes/celery-beat/deployment.yml:46]: In the mean time some docs ^^
https://sysdig.com/wp-content/uploads/2019/01/kubernetes-security-guide.pdf

chapter 4 and 5

File: kubernetes/celery-beat/deployment.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      volumes:
+      - name: django-fileserver-storage
+        persistentVolumeClaim:
+          claimName: fileserver-claim
+      serviceAccountName: {{.Values.serviceAccounts.cloudSqlStorage}}
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:51]: EOF

File: kubernetes/celery-beat/deployment.yml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      - name: cloud-sql-proxy
+        image: {{.Values.database.cloudsqlimage}}
+        command:
+        - "/cloud_sql_proxy"
+        - "-ip_address_types=PRIVATE"
+        - "--instances={{.Values.database.instanceConnectionName}}=tcp:{{.Values.database.dbPort}}"
+        securityContext:
+          runAsNonRoot: true
+      volumes:
+      - name: django-fileserver-storage
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:48]: Nice, well done

File: kubernetes/django/deployment.yml [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      labels:
+        app: django
+    spec:
+      containers:
+      - name: django
+        image: {{.Values.django.image.repo}}:{{.Values.django.image.tag}}
+        ports:
+        - name: django-port
+          containerPort: {{.Values.django.containerPort}}
+          protocol: TCP
```
Comment (tychota) [inline kubernetes/django/deployment.yml:23]: nice

File: kubernetes/celery-beat/deployment.yml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  selector:
+    matchLabels:
+      app: celery-beat
+  template:
+    metadata:
+      labels:
+        app: celery-beat
+    spec:
+      containers:
+      - name: celery-beat
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:20]: [Importance 2/10]

Ideally we also needs some limits.
https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:20]: [Importance 4/10]

Ideally we should know if the pod is up and restart if not
we can use https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

File: kubernetes/celery-worker/deployment.yml [hunk 12]
```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
```
Comment (tychota) [inline kubernetes/celery-worker/deployment.yml:1]: Same than beat

File: kubernetes/django/deployment.yml [hunk 13]
```diff
@@ -0,0 +1,69 @@
+apiVersion: apps/v1
```
Comment (tychota) [inline kubernetes/django/deployment.yml:1]: Same than beat

File: kubernetes/django/deployment.yml [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+            # POSTGRES_USER and POSTGRES_PASSWORD
+            name: postgres-env-secrets
+        env:
+        - name: POSTGRES_HOST
+          value: {{.Values.database.dbIpAddressWithinContainer}}
+        - name: POSTGRES_DB
+          value: {{.Values.database.dbName}}
+        - name: POSTGRES_PORT
+          value: {{.Values.database.dbPort}}
+        readinessProbe:
```
Comment (tychota) [inline kubernetes/django/deployment.yml:38]: Nice

File: kubernetes/ingress.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
+apiVersion: networking.k8s.io/v1beta1
```
Comment (tychota) [inline kubernetes/ingress.yml:1]: nice

File: kubernetes/nginx/deployment.yml [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+          periodSeconds: 20
+          timeoutSeconds: 3
+          failureThreshold: 5
+        volumeMounts:
+        - name: nginx-fileserver-storage
+          mountPath: /data
+      volumes:
+      - name: nginx-fileserver-storage
+        persistentVolumeClaim:
+          claimName: fileserver-claim
```
Comment (tychota) [inline kubernetes/nginx/deployment.yml:45]: EOF

File: kubernetes/celery-beat/deployment.yml [hunk 17]
```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: celery-beat-deployment
+  labels:
+    app: celery-beat
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:6]: [Importance 6/10]

We should dod this trick https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments

File: kubernetes/celery-beat/deployment.yml [hunk 18]
```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: celery-beat-deployment
+  labels:
```
Comment (tychota) [inline kubernetes/celery-beat/deployment.yml:5]: [Importance 8/10]

We are missing labels/annotation used by helm to see what it manage

https://helm.sh/docs/chart_best_practices/labels/

**Comment by tychota** [comment]
You did well, that is a good first step.

---

For it to work you need to respect helm structure.
You can read more here: https://helm.sh/docs/chart_template_guide/getting_started/

For instance it would make sense to have threehelms chart for now.
- one with django and celery beat/worker, and associated service / ingress / secrets / configmap
- one with nginx, and associated service / ingress
- (one with redis (clusterised, maybe), and associated service / ingress)

For redis we may  want to use an operator, like https://github.com/spotahome/redis-operator (there is a spooky  or the one from redis itself https://redislabs.com/blog/redis-enterprise-operator-kubernetes/
We should decide if we want or not: I'm not sure, open to discussion https://medium.com/swlh/production-checklist-for-redis-on-kubernetes-60173d5a5325
If you want you can read more about the benefir of operators https://coreos.com/operators/ and https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ and https://blog.container-solutions.com/kubernetes-operators-explained but the rule of thumb is that stateful (eg database) running in kubernetes needs an operator.

--- 

> Redis is missing from this push, but I think mostly everything else is here aside from secrets?

secrets shouldn't be part of the helm template but created separately

you also miss the "Job" part in django 

https://itnext.io/database-migrations-on-kubernetes-using-helm-hooks-fb80c0d97805

 ---

> the expected workflow for using these files

First we are going to build images and upload them to registry

Then install template in cluster using something like in https://helm.sh/docs/chart_template_guide/getting_started/#a-first-template

```bash
helm install katago-server ./katago-server
```

Then later, after upgrading the version in values.yaml

```bash
helm upgrade katago-server ./katago-server
```

https://helm.sh/docs/intro/using_helm/#helm-upgrade-and-helm-rollback-upgrading-a-release-and-recovering-on-failure

---

> versioning and tagging and upload of images

We can use git destribe for the version

Eg

```
docker build -t grc.io/org/proj/katago-server:$(git describe --abbrev=7 --tags --always --first-parent) -t katago-server:latest ./path/to/katago/dockerfile
docker push grc.io/org/proj/katago-server:$(git describe --abbrev=7 --tags --always --first-parent)
docker push grc.io/org/proj/katago-server:latest
```

---

>  how secrets work, how internal services communication and hostname works for redis and celery

not sure what you mean

Secret:
- are crd (custom resource) for kube, store normally in etcd as clear base64 but in our cluster we use google kms
- https://kubernetes.io/docs/concepts/configuration/secret/#security-properties
- https://cloud.google.com/kubernetes-engine/docs/how-to/encrypting-secrets

Communication kubernetes:
- this is the detailled way kubernetes networking works by default https://sookocheff.com/post/kubernetes/understanding-kubernetes-networking-model/
- the networking itself is an abstraction so you can change the way it works: it use a CNi (and you can even define your own CNI for networking if you are crazy https://www.altoros.com/blog/kubernetes-networking-writing-your-own-simple-cni-plug-in-with-bash/): for instance cilium use e-bpf https://www.youtube.com/watch?v=_Iq1xxNZOAo and calisato use l2 and BGP https://docs.projectcalico.org/networking/determine-best-networking
- google does the hard job for you: by default like normal (veth, bridge, arp layer 2) https://cloud.google.com/kubernetes-engine/docs/concepts/network-overview and since a few month you can opt in cillium https://cloud.google.com/blog/products/containers-kubernetes/bringing-ebpf-and-cilium-to-google-kubernetes-engine
- basically you don't have to care how it is done except you need some really specif stuff

DNS in kubernetes:
- kube setup a dns for you: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/
- https://cloud.google.com/kubernetes-engine/docs/concepts/service-discovery

File: kubernetes/celery-beat/deployment.yml (hunk 8 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml:51]: Thanks, will fix in upcoming commit.

File: compose/production/django/entrypoint [hunk 19]
```diff
@@ -14,7 +14,7 @@ if [ -z "${POSTGRES_USER}" ]; then
     base_postgres_image_default_user='postgres'
     export POSTGRES_USER="${base_postgres_image_default_user}"
 fi
-export DATABASE_URL="postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}"
+export DATABASE_URL="${POSTGRES_HOST}://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}"
```
Comment (tychota) [inline compose/production/django/entrypoint:17]: revert to postgres://

File: kubernetes/celery-beat/deployment.yml (hunk 6 above)
Comment (tychota) [reply kubernetes/celery-beat/deployment.yml:44]: I can look at it ?

File: kubernetes/celery-beat/deployment.yml (hunk 1 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml:4]: I did a partial step - I moved everything out of default namespace but I didn't go with the complication of having different namespaces for different groups of pods.

File: kubernetes/celery-beat/deployment.yml (hunk 2 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml:14]: Commented why in the file as discussed, thanks.

File: kubernetes/celery-beat/deployment.yml (hunk 3 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml:26]: Current thing is fine, don't want to bother reworking images. :)

File: kubernetes/celery-beat/deployment.yml (hunk 11 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml:24]: The links we looked at only works for celeryworkers, it doesn't serve as a check for celerybeat. (I tested this out on the test prod server manually killing or suspending celerybeat and seeing what happened).

So I added the liveness probe for the worker, but for celerybeat for now I give up and just let it fail if it fails.

File: compose/production/django/entrypoint (hunk 19 above)
Comment (lightvector) [reply compose/production/django/entrypoint]: done

File: kubernetes/celery-beat/deployment.yml (hunk 18 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml]: done

File: kubernetes/celery-beat/deployment.yml (hunk 6 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml]: I did it, thanks.

File: kubernetes/celery-beat/deployment.yml (hunk 4 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml]: Done

File: kubernetes/celery-beat/deployment.yml (hunk 17 above)
Comment (lightvector) [reply kubernetes/celery-beat/deployment.yml]: I think this is done now, thanks.

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+    matchLabels:
+      app: celery-flower
+      {{- include "katago-server.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app: celery-flower
+        {{- include "katago-server.selectorLabels" . | nindent 8 }}
+      annotations:
+        checksum/django_env_secrets: {{ include (print $.Template.BasePath "/env/django_env_secrets.yaml") . | sha256sum }}
```
Comment (tychota) [inline kubernetes/katago-server/templates/celery-flower/deployment.yaml:22]: How does that work ? does it require to modify the template ?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+          tcpSocket:
+            port: celery-flower-port
+          initialDelaySeconds: 15
+          periodSeconds: 20
+          timeoutSeconds: 3
+          failureThreshold: 5
+        resources:
+          limits:
+            cpu: 100m
+        volumeMounts:
```
Comment (tychota) [inline kubernetes/katago-server/templates/celery-flower/deployment.yaml:60]: Shall we mount the main storage ?
I guess we can juste use a small amount of cloud storage

Eg
https://cloud.google.com/kubernetes-engine/docs/how-to/stateful-apps?hl=fr#creating_a_statefulset

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+        - name: django-fileserver-storage
+          mountPath: /data
+          subPath: {{.Values.fileserver.subPath}}
+      - name: cloud-sql-proxy
```
Comment (tychota) [inline kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: Also not sure we need that. 

Flower should not connect to DB

File: kubernetes/katago-server/values.yaml [hunk 23]
```diff
@@ -19,6 +19,10 @@ celeryWorker:
   replicaCount: 1
 celeryBeat:
   replicaCount: 1
+celeryFlower:
```
Comment (tychota) [inline kubernetes/katago-server/values.yaml:22]: nice

File: kubernetes/katago-server/templates/ingress.yaml [hunk 24]
```diff
@@ -22,4 +22,8 @@ spec:
       - path: /
         backend:
           serviceName: django-service
-          servicePort: {{.Values.django.servicePort}}
\ No newline at end of file
+          servicePort: {{.Values.django.servicePort}}
+      - path: /
```
Comment (tychota) [inline kubernetes/katago-server/templates/ingress.yaml:26]: That looks false to me. You can't have two same path for the same host.

I will create another host eg `flower.katagotraining.org`

File: kubernetes/katago-server/templates/django/migrate.yaml [hunk 25]
```diff
@@ -0,0 +1,49 @@
+apiVersion: batch/v1
+kind: Job
```
Comment (tychota) [inline kubernetes/katago-server/templates/django/migrate.yaml:2]: nice

File: kubernetes/katago-server/templates/django/migrate.yaml [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+metadata:
+  name: django-migrate-job
+  labels:
+    {{- include "katago-server.labels" . | nindent 4 }}
+spec:
+  backoffLimit: 2
+  ttlSecondsAfterFinished: 900
```
Comment (tychota) [inline kubernetes/katago-server/templates/django/migrate.yaml:14]: Why such an hight number of seconds ? Maybe it is ok but i don't understand the use case

File: kubernetes/katago-server/templates/django/migrate.yaml [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+        - configMapRef:
+            # GCP_DATABASE_INSTANCE and POSTGRES_PORT
+            name: postgres-env-config
+        securityContext:
+          runAsNonRoot: true
+      volumes:
+      - name: django-fileserver-storage
+        persistentVolumeClaim:
+          claimName: fileserver-claim
+      serviceAccountName: {{.Values.serviceAccounts.cloudSqlStorage}}
```
Comment (tychota) [inline kubernetes/katago-server/templates/django/migrate.yaml]: EOF

File: kubernetes/katago-server/templates/celery-beat/deployment.yaml [hunk 28]
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: serverspace
   labels:
     app: celery-beat
+    app.kubernetes.io/component: celery-beat
```
Comment (tychota) [inline kubernetes/katago-server/templates/celery-beat/deployment.yaml:8]: nice

**Review by tychota — APPROVED** [review]

**Comment by tychota** [comment]
image

File: kubernetes/katago-server/templates/django/migrate.yaml (hunk 26 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/django/migrate.yaml:14]: I want to be able to see the container for a while after the job inside it failed, to make it easier to inspect it and look at logs and so on. :)

File: kubernetes/katago-server/templates/ingress.yaml (hunk 24 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/ingress.yaml:26]: You can't have two paths on the same host even if they are different ports?
Flower is on port 5555, right? This is how we had it in docker-compose production.yml too, right?. We have path "/" with port 80 or port 443 go to django, except that if also has the prefix "/media" it goes nginx. And we had path "/" on port 5555 to to flower?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 20 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:22]: I am simply doing the recommendation that *you* linked! Right?
https://github.com/katago/katago-server/pull/44#discussion_r517000159

My understanding of workflow was that we will have a secrets file env/django_env_secrets.yaml sitting on disk (unencrypted while it is on disk?) that we deploy, the same way we have a secrets file in docker compose sitting (unencrypted) on disk that is mounted by production.yml into the environment variables for a container.

This uses the mechanism in the link you have - if the secrets file changes, then the sha256sum will change, and so this annotation will change, and then helm will know that this deployment needs to be updated and to do a rolling restart properly.

Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:22]: Is there a different workflow you expected?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 21 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:60]: In production.yml we simply used exactly mirrored settings for all of the celery containers, compared to the django container. Same mounts, same environment variables, etc. I'm simply doing that here, since I wasn't sure what would be safe to be different or not. There should be no harm, right?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 22 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: In production.yml we simply used exactly mirrored settings for all of the celery containers, compared to the django container. Same mounts, same environment variables, etc. I'm simply doing that here, since I wasn't sure what would be safe to be different or not. There should be no harm, right?

Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: But okay, if you are sure flower doesn't need it, I can remove it. :)

Comment (tychota) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: Normally it shouldn't, if it does it is a bug for me. Meaning I'm not actually sure it doesn't need.

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 20 above)
Comment (tychota) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:22]: makes sense.

I wasn't planing to deploy secrets with helm so I was thinking of only doing checksum for configmap (that can be deploy with helm, same as https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments).

Also based on

image

I'm not sure we need it for env variables (sorry if I made you do useless stuff).

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 21 above)
Comment (tychota) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:60]: Except if there is a bug in flower and someone can delete all file from the flower container.

Ideally I would try in production to just give what container needs.

File: kubernetes/katago-server/templates/ingress.yaml (hunk 24 above)
Comment (tychota) [reply kubernetes/katago-server/templates/ingress.yaml:26]: I'm almost certain it won't work: https://cloud.google.com/kubernetes-engine/docs/concepts/ingress

The ingress itself is a loadblancer listenning on only 80 and 443 (like traefik by default)

image

to be able to reach `whatever.url.com:5555` we do not only need the internal pod to listen on 5555 but also the ingress.
On traefik we opened a new endpoint for this.

On GCP i don't know how to do, with defult ingress controller. (with another ingress controller, you can do this for instance https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/)

My point isnon standard port like 5555 is non standard so we should prefer having two hosts

File: kubernetes/katago-server/templates/django/migrate.yaml (hunk 26 above)
Comment (tychota) [reply kubernetes/katago-server/templates/django/migrate.yaml:14]: ok for me

File: kubernetes/katago-server/templates/ingress.yaml (hunk 24 above)
Comment (tychota) [reply kubernetes/katago-server/templates/ingress.yaml:26]: Also we have not configured https yet: here is how to do https://cloud.google.com/kubernetes-engine/docs/how-to/managed-certs

Comment (lightvector) [reply kubernetes/katago-server/templates/ingress.yaml:26]: Maybe we should just use nginx ingress controller then?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 21 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:60]: Okay so technically in the docker compose config we should also change it to not mount the database too, right?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 22 above)
Comment (lightvector) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: Okay so technically in the docker compose config we should also change it to not mount the database and the storage too, right?

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 21 above)
Comment (tychota) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:60]: I think yes, but not sure if it works

File: kubernetes/katago-server/templates/celery-flower/deployment.yaml (hunk 22 above)
Comment (tychota) [reply kubernetes/katago-server/templates/celery-flower/deployment.yaml:64]: I think yes, but not sure if it works

File: kubernetes/katago-server/templates/ingress.yaml (hunk 24 above)
Comment (tychota) [reply kubernetes/katago-server/templates/ingress.yaml:26]: not sure we really need to expose a custom port

we can have

`katagotraining.com -> django`
`flower.katagotraining.com ->flower`

that would be more clean for sure

File: kubernetes/katago-server/templates/frontendconfig.yaml [hunk 29]
```diff
@@ -0,0 +1,8 @@
+apiVersion: networking.gke.io/v1beta1
+kind: FrontendConfig
+metadata:
+  name: katago-server-frontend-config
+spec:
+  redirectToHttps:
+    enabled: true
+    responseCodeName: MOVED_PERMANENTLY_DEFAULT
```
Comment (tychota) [inline kubernetes/katago-server/templates/frontendconfig.yaml:8]: EOF
